code/modules/projectiles/guns/energy/modular.dm

```diff
@@ -185,6 +188,9 @@
 	if(modulator)
 		modulator.forceMove(A)
 		modulator = null
+	if(pin)
+		pin.forceMove(A)
```

Comment:
I'm pretty sure there is a proc for removing a pin from a gun. Use that. Or you should at least do `pin.gun = null`

Reply:
I am sure it is already done in the line bellow.